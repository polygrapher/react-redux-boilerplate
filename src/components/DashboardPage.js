import React from 'react';

const DashboardPage = () => (
    <div>
        <h1>Dashboard content</h1>
    </div>
);

export default DashboardPage;