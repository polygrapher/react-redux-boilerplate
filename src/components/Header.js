import React from 'react';
import {Link} from 'react-router-dom'
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
import {bindActionCreators} from 'redux';

export const Header = ({startLogout}) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link to="/dashboard" className="header__title">
                    <h1>Boilerplate React App</h1>
                </Link>
                <button onClick={startLogout} className="button button--link">Log out</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => bindActionCreators({startLogout}, dispatch);

export default connect(undefined, mapDispatchToProps)(Header);