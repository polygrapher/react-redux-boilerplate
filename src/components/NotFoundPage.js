import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <div>404! - Go to <Link to="/">Home</Link> page</div>
);

export default NotFoundPage;