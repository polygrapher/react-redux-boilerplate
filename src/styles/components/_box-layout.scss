.box-layout {
    display: flex;
    justify-content: center;
    align-items: center;
    background: url("/images/bg.jpg") center / cover no-repeat;
    height: 100vh;
    width: 100vw;

    &__box {
        background-color: fade-out($color: #fff, $amount: .15);
        border-radius: 3px;
        text-align: center;
        padding: $l-size $m-size;
        width: 25rem;
    }

    &__title {
        margin: 0 0 $m-size 0;
        line-height: 1;
    }
}