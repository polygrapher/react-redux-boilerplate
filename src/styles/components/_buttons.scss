.button {
    display: inline-block;
    line-height: 1;
    padding: $s-size;
    border: none;
    color: #fff;
    font-size: $font-size-large;
    font-weight: 300;
    background-color: $blue;
    text-decoration: none;

    &--link {
        background: none;
    }

    &--secondary {
        background-color: #888888;
    }
}