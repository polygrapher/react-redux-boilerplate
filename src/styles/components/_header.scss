.header {
    background-color: $dark-blue;

    &__title {
        color: #fff;
        margin: 0;
        text-decoration: none;
    }

    &__content {
        display: flex;
        padding: $s-size 0;
        justify-content: space-between;
        align-items: center;
    }
}