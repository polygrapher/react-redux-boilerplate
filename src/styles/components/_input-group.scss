.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $l-size;
    @media screen and (min-width: $desktop-breakpoint) {
        flex-direction: row;
    }

    &__item {
        margin-bottom: $s-size;
        @media screen and (min-width: $desktop-breakpoint) {
            margin: 0 $s-size 0 0;
        }
    }
}