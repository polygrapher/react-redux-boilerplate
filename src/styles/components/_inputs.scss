.text-input {
    padding: $s-size;
    height: 50px;
    border: 1px solid #cacccd;
    font-weight: 300;
    $font-size: $font-size-large;
}

.select {
    @extend .text-input;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}