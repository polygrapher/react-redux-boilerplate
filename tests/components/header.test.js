import React from 'react';
import {shallow} from 'enzyme';
import {Header} from '../../src/components/Header';
import { start } from 'repl';


let wrapper, startLogout;

beforeEach(() => {
    startLogout = jest.fn(),
    wrapper = shallow(
        <Header
            startLogout={startLogout}
        />
    );
});

test('Header renders correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot('Expensify');
});

test('startLogout action is called on logout button click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogout).toBeCalled();
});
