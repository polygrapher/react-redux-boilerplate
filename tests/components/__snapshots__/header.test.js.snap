// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly: Expensify 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate React App
        </h1>
      </Link>
      <button
        className="button button--link"
      >
        Log out
      </button>
    </div>
  </div>
</header>
`;
